CONTRIBUTING_CODE.md

```diff
@@ -0,0 +1,302 @@
+
+### Forking the repo
+
+In order to do pull requests, begin by forking the repository on GitHub. 
+Then, clone your fork to a local machine.
+
+```shell
+$ git clone git@github.com:MyGithubAccount/probnum.git
+```
+Any code changes should be done in your fork. Github as official guides on [how to fork](https://docs.github.com/en/github/getting-started-with-github/fork-a-repo) 
+and [how to pull request](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request).
+
+### Multiple remotes
+In order to keep your fork up-to-date with the main repository, one convenient
+way to do so is to set up [multiple remotes](https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes)
+for `git`.
+For example, this way you can always keep the master branch of your forked repository
+up-to-date with the main repository and you can easily `git merge` recent changes into
+your feature branch. To set up the main ProbNum repository as a secondary remote run
+
+```shell
+$ cd probnum
+$ git remote add upstream https://github.com/probabilistic-numerics/probnum
+```
+
+Now in addition to `origin`, which is your fork of ProbNum, you have access to the main
+repository named `upstream`. Check your remotes are set up properly by executing the following command.
+
+```shell
+$ git remote -v
+origin	https://github.com/MyGithubAccount/probnum.git (fetch)
+origin	https://github.com/MyGithubAccount/probnum.git (push)
+upstream	https://github.com/probabilistic-numerics/probnum (fetch)
+upstream	https://github.com/probabilistic-numerics/probnum (push)
+```
+Fetch and checkout the master branch of ProbNum. 
+Since there exists already a local `master` branch tracking the master branch of your fork, we will name this one
+`probnum-master` which tracks the master branch of `upstream`.
+
+```shell
+$ git fetch upstream
+$ git checkout -b probnum-master upstream/master
+```
+
+### Code changes
+Next, create a new branch describing the feature you would like to implement. 
+```shell
+$ git checkout -b my-new-feature
+```
+You can now get started writing code in your new branch. 
+
+Commit your changes to this local branch. Make sure to keep the best practices regarding code quality in mind. 
+Next, set up a branch on your fork which is tracked by your local branch. 
+```shell
+git push --set-upstream origin my-new-feature
+```
+Once in a while, merge changes of the upstream master branch into your local `my-new-feature` branch.
+```shell
+$ git checkout probnum-master
+$ git pull
+$ git checkout my-new-feature
+$ git merge probnum-master
+```
+### Open a pull-request
+Once you are happy with your PR, make sure your branch is up-to-date with upstream `master`, and re-run all tests.
+Then, from your local branch, push the changes to your remote branch
+```shell
+$ git push
+```
+
+and do the PR via the github interface. There, briefly explain the changes as well as how they were tested. That's it!
```

Comment:
and open the PR via the GitHub interface.

Is it really necessary to explain how the changes were tested? Any tests are part of the PR and the coverage / test results are automatically reported by the CI. It seems that this is somewhat unnecessary.

Reply:
hm, well, it's not really necessary I suppose. I guess it creates a good habit of testing before creating the PR. I can change the sentence to "Please try running the test suite with tox or pytest before creating the PR." Does that sounds OK?

I wouldn't encourage PRs that do not complete at least some tests beforehand (eg for a single python version) as PRs shouldn't be used as debugging tool.

Reply:
typo is fixed